### PR TITLE
[cli] Fix analyze validators fetching, when we hit 1000 limit for API

### PR DESCRIPTION
**Pull request by igor-aptos**
### Test Plan
cargo run -p aptos -- node analyze-validator-performance --analyze-mode=all --url=https://ait3.aptosdev.com --start-epoch=54 --end-epoch=57

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3813)

### CODE REVIEW

File: crates/aptos/src/node/analyze/fetch_metadata.rs [hunk 1]
```diff
@@ -72,6 +72,25 @@ impl FetchMetadata {
...
+            let limit = std::cmp::min(1000, last - cursor);
+            let mut current = client
+                .get_transactions(Some(cursor), Some(limit as u16))
+                .await?
+                .into_inner();
+            result.append(&mut current);
+            cursor += limit
+        }
+        Ok(result)
+    }
```
Comment (gregnazario) [inline crates/aptos/src/node/analyze/fetch_metadata.rs:92]: So this is just pinning the maximum page size to 1000, and parsing all the transactions at that rate?

Comment (igor-aptos) [reply crates/aptos/src/node/analyze/fetch_metadata.rs:92]: Yes. 

This is needed just for finding validator set for the epoch change. 

We go over all NewBlockEvents, and search within transactions of the last block in the epoch to find the validator set

**Comment by igor-aptos** [comment]
added voting power calculation to this diff.

**Review by davidiw — APPROVED** [review]
good for me, @gregnazario ?

**Review by ibalajiarun — APPROVED** [review]
LGTM.

**Pull request merged**